src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/AutoDisenchanter.java

```diff
@@ -73,8 +77,13 @@ protected MachineRecipe findNextRecipe(BlockMenu menu) {
                 int amount = 0;
 
                 for (Map.Entry<Enchantment, Integer> entry : item.getEnchantments().entrySet()) {
-                    enchantments.put(entry.getKey(), entry.getValue());
-                    amount++;
+                    if (entry.getValue() <= enchantLevelLimit.getValue() || enchantLevelLimit.getValue() < 1) {
+                        enchantments.put(entry.getKey(), entry.getValue());
+                        amount++;
+                    } else {
+                        if (!menu.toInventory().getViewers().isEmpty()) {
+                            menu.toInventory().getViewers().forEach(p -> SlimefunPlugin.getLocalization().sendMessage(p, "messages.above-limit-level", true, (message -> message.replace("%level%", enchantLevelLimit.getValue().toString()))));}
```

Comment:
1. Never use `.forEach(Consumer)`.
2. Sending a message to all inventory viewers is rather akwards, not something that we do here. I reckon it would make more sense to utilize the "progress bar" slot in the middle for this and update that instead.

Reply:
Have no idea how AContainer change lore of progress bar, Can I modify getProgressBar() directly?

Reply:
Done.